src/Scene/AnimationPlayer.js

```diff
@@ -0,0 +1,203 @@
+var _DEBUG = null;
+
+if (debugAnimation) {
+    _DEBUG = function (message, animation) {
+        /* eslint-disable no-alert, no-console */
+        if (animation) {
+            console.info('Animation ', message, ' : ', animation.name);
+        }
+        /* eslint-enable no-alert, no-console */
+    };
+} else {
+    _DEBUG = function () {};
+}
+
+// Private functions
+// stop timer and re-init parameter
+var resetTimer = function () {
+    if (this.id) {
+        clearInterval(this.id);
+        this.id = undefined;
+    }
+
+    this.keyframe = 0;
+};
+
+// finish animation and re-init parameter
+var finishAnimation = function () {
+    resetTimer.bind(this)();
+    this.animation = null;
+    this.resolve();
+};
+
+var setPlayerState = function (state) {
+    this.state = state;
+    _DEBUG(debugMsg[state], this.animation);
+};
+
+// Public functions
+AnimationPlayer.prototype.isPlayed = function () {
+    return this.state === PLAYER_STATE.PLAY;
+};
+
+AnimationPlayer.prototype.isStopped = function () {
+    return this.state === PLAYER_STATE.STOP;
+};
+
+AnimationPlayer.prototype.isEnded = function () {
+    return this.state === PLAYER_STATE.END;
+};
+
+/**
+ * { Start animation }
+ *
+ * @param      {Animation} The animation to play
+ * @return     {Promise}  Promise is resolved when animation is stopped or finished
+ */
+AnimationPlayer.prototype.play = function (animation) {
+    this.animation = animation;
+    setPlayerState.bind(this)(PLAYER_STATE.PLAY);
+    resetTimer.bind(this)();
+    this.id = setInterval(this.frame.bind(this), INTERVAL_TIME);
+    return new Promise((r) => { this.resolve = r; });
+};
+
+/**
+ * { Stop current animation }
+ *
+ * @return  {Promise}  Promise is resolved when animation is stopped or finished
+ */
+AnimationPlayer.prototype.stop = function () {
+    setPlayerState.bind(this)(PLAYER_STATE.STOP);
+    finishAnimation.bind(this)();
+    this.dom.dispatchEvent(this.stopEvent);
+    return this.promise;
+};
+
+/**
+ * { Pause current animation }
+ */
+AnimationPlayer.prototype.pause = function () {
+    setPlayerState.bind(this)(PLAYER_STATE.PAUSE);
+    clearInterval(this.id);
+    this.id = undefined;
+};
+
+/**
+ * { this function is executed with each frame }
+ */
+AnimationPlayer.prototype.frame = function () {
+    if (this.keyframe === this.animation.duration) {
+        setPlayerState.bind(this)(PLAYER_STATE.END);
+        finishAnimation.bind(this)();
+        this.dom.dispatchEvent(this.endEvent);
+    }
+    else {
+        if (this.animation) {
+            this.animation.animate(this.keyframe);
+        }
+        this.keyframe++;
+        this.dom.dispatchEvent(this.frameEvent);
+    }
+};
+
+/**
+ * { Animation }
+ *
+ * @class      Animation (name)
+ * @param      {Number}  duration  The animation's duration
+ * @param      {Object}  root      The root object to animate
+ * @param      {String}  name      The animation's name
+ */
+function Animation(duration, root, name) {
```

Comment:
> Why do you need this class?

There will be `keyFrameAnimation` inherit from `Animation` 

> `{ 30, elem, 'name', animationFn}`

Ok

> remove `Animation` and `AnimatedExpression` declarations entirely and replace `this.animation.animate(this.keyframe)` by `this.animation.expression(this.animation.root, keyFrame / this.animation.duration)`

I would prefer not to

> all the AnimatedExpression declaration on GlobeControl would just be some plain objects that would be stored on this.animation of AnimationPlayer

I don't understand

> Document what animation should look like in `AnimationPlayer.play`

Ok